OrbitLinuxTracing/PerfEvent.h

```diff
@@ -185,6 +185,58 @@ struct dynamically_sized_perf_event_stack_sample {
       : stack{dyn_size} {}
 };
 
+struct dynamically_sized_perf_event_call_chain_sample {
+  struct dynamically_sized_perf_event_call_chain {
+    uint64_t dyn_size;
+    std::unique_ptr<uint64_t[]> data;
+
+    explicit dynamically_sized_perf_event_call_chain(uint64_t dyn_size)
+        : dyn_size{dyn_size},
+          data{make_unique_for_overwrite<uint64_t[]>(dyn_size)} {}
+  };
+
+  perf_event_header header;
+  perf_event_sample_id_tid_time_streamid_cpu sample_id;
+  dynamically_sized_perf_event_call_chain call_chain;
+
+  explicit dynamically_sized_perf_event_call_chain_sample(uint64_t dyn_size)
+      : call_chain{dyn_size} {}
+};
+
+class SampleFPPerfEvent : public PerfEvent {
```

Comment:
Rename this to SampleCallChainPerfEvent. This is more consistent with dynamically_..._call_chain_sample. Also, I don't think the method of how the call chain was obtained should be reflected in the name. Finally, this gets rid of the abbreviation "FP".

Reply:
That name is better. Thanks for the suggestion. I'll take care, that the "FP" abbreviation will not make it into master.

Reply:
Indeed, but I think it should be `CallchainSamplePerfEvent` (don't split "callchain" in two words, and "callchain" should come before "sample" as it acts as an adjective).

Reply:
Thanks @dpallotti for the better suggestion. Agree we should follow the official perf conventions whenever possible.

Reply:
Thats right. Didn't looked in the manpage when renaming.